vlm/src/main/scala/geotrellis/contrib/vlm/gdal/Patterns.scala

```diff
@@ -0,0 +1,21 @@
+package geotrellis.contrib.vlm.gdal
+
+import scala.util.matching.Regex
+
+
+/*
+ * Contains the different Regexs needed to parse
+ * the paths given to GDAL to read.
+ */
+object Patterns {
```

Comment:
As a general comment, not a request for change, I would say that this parsing would be much easier to think about and more durable if you were to use RegEx groups and tried to parse the whole URI as a whole unit in a single RegEx path.

Reply:
I wasn't aware we could do that in Scala. I'll look and see if that would simplify things.